### DISCUSSION TITLE
Using next/image with CardMedia in material-ui

### DISCUSSION THREAD

**Discussion by gdowmont** [vercel/next.js#23352] [Help] [answered]
I have defined the src in CardMedia as and that works as expected scaling the image to the size of CardMedia:

```js
<CardMedia
  className={classes.cardMedia}
  image={recipe.images.small_image_url}
  title={recipe.title}
/>
```
I have tried using next/image and it displays the image but it doesn't place it inside CardMedia parent and just uses the full page instead.
```js
<CardMedia className={classes.cardMedia} title={recipe.title}>
  <Image
    src={recipe.images.small_image_url}
    layout="fill"
    objectFit="contain"
  />
</CardMedia>
```

If use img, it contains it within the CardMedia:
```js
<CardMedia className={classes.cardMedia} title={recipe.title}>
  <img src={recipe.images.small_image_url} />
</CardMedia>
```

Has anyone got any idea if I can do anything so next/image automatically follows the size set for CardMedia?

**Comment by akellbl4** [comment, marked as answer]
Hi,
I can suggest two different approaches.
Do you have a fixed size of `CardMedia`?
If you have:
```jsx
<CardMedia className={classes.cardMedia} title={recipe.title}>
  <Image
    src={recipe.images.small_image_url}"
    width={200}
    height={200}
    objectFit="contain" // or objectFit="cover"
  />
</CardMedia>
```

If you don't have:
```jsx
<CardMedia className={classes.cardMedia} title={recipe.title}>
  <div style={{ position: 'relative', width: '100%', height: '100%' }}>
    <Image
      src={recipe.images.small_image_url}"
      layout="fill"
      objectFit="contain" // or objectFit="cover"
    />
  </div>
</CardMedia>
```

**Reply by gdowmont** [reply]
Thank you, @akellbl4

This worked exactly how I needed with `cover`:
```js
<CardMedia className={classes.cardMedia} title={recipe.title}>
  <div style={{ position: 'relative', width: '100%', height: '100%' }}>
    <Image
      src={recipe.images.small_image_url}"
      layout="fill"
      objectFit="cover"
    />
  </div>
</CardMedia>
```

**Reply by mohamad-ghasemilou** [reply]
this is throwing an error because the CardMedia missing component attribute

**Comment by StarAzure** [comment]
Does not work with mui5. layout and objectfit are deprecated.

**Reply by jilherme** [reply]
```
<CardMedia sx={{ position: 'relative' }}>
        <Image
          src={image}
          width={320}
          height={200}
          alt=""
          style={{ objectFit: 'cover' }}
        />
</CardMedia>
```
      
It worked for me. Better late than never

**Comment by redimongo** [comment]
> <CardMedia sx={{ position: 'relative' }}>
>         <Image
>           src={image}
>           width={320}
>           height={200}
>           alt=""
>           style={{ objectFit: 'cover' }}
>         />
> </CardMedia>

Could not get this to work
